src/crypto/equihash.cpp

```diff
@@ -303,6 +303,28 @@ bool Equihash<N,K>::BasicSolve(const eh_HashState& base_state,
     return false;
 }
 
+bool IsProbablyDuplicate(std::shared_ptr<eh_trunc> indices, size_t lenIndices)
+{
+    bool checked_index[lenIndices] = {false};
+    for (int z = 0; z < lenIndices; z++) {
+        if (!checked_index[z]) {
+            for (int y = z+1; y < lenIndices; y++) {
+                if (!checked_index[y] && indices.get()[z] == indices.get()[y]) {
+                    // Pair found
+                    checked_index[y] = true;
+                    checked_index[z] = true;
+                    break;
+                }
+            }
+        }
+    }
+    bool is_probably_duplicate = true;
+    for (int z = 0; z < lenIndices && is_probably_duplicate; z++) {
+        is_probably_duplicate &= checked_index[z];
+    }
+    return is_probably_duplicate;
```

Comment:
Suggested optimisation/simplification: maintain a count of the number of indices for which ``checked_index`` is true (incremented by 2 at line 316). Then return ``count_checked == lenIndices`` instead of the ``is_probably_duplicate`` loop.